Nef_S2/include/CGAL/Nef_S2/Sphere_circle.h

```diff
@@ -61,7 +63,7 @@ Sphere_circle(const Sphere_point<R>& p, const Sphere_point<R>&q)
 such that a walk along |\Mvar| meets $p$ just before the shorter segment
 between $p$ and $q$. If $p$ and $q$ are antipodal of each other then we
 create any great circle that contains $p$ and $q$.}*/
-{ Point_3 p1(0,0,0), p4 = CGAL::ORIGIN + ((Base*) this)->orthogonal_vector();
+{ Point_3 p1(0,0,0), p4 = CGAL::ORIGIN + Base::orthogonal_vector();
   if ( p != q.antipode() ) {
     if (R_().orientation_3_object()(p1,Point_3(p),
```

Comment:
No idea if this line shows up often in vtune, but maybe it makes sense to have an orientation predicate for the case that one point is the origin.

Reply:
That would be the same as the `Orientation_3::operator()(Vector_3, Vector_3, Vector_3)`, but with points instead of vectors.

I think `Sphere_point<K>` should actually derive from `K::Vector_3`, and not `K::Direction_3` and not `K::Point_3`, and that would make the code simpler.

Reply:
@lrineau That makes sense, a point on the sphere is actually a vector from the origin of the sphere towards the point on the surface of the sphere.